src/jit/lclvars.cpp

```diff
@@ -3292,6 +3292,13 @@ void Compiler::lvaSortByRefCount()
         {
             varDsc->lvTracked = 0;
         }
+#if defined(JIT32_GCENCODER) && defined(WIN64EXCEPTIONS)
+        else if (lvaIsOriginalThisArg(lclNum))
+        {
+            // For x86/Linux, we cannot have this pointer in tracked variables, so we set this pointer untracked
+            varDsc->lvTracked = 0;
```

Comment:
@BruceForstall I was going to change the condition to:

```cpp
        else if (lvaIsOriginalThisArg(lclNum) && lvaKeepAliveAndReportThis())
```

However we need something like `lvaKeepAliveAndReportThis()` from VM. Could I ask how to get a method info from vm(`GetInstance():eetwain.cpp`)? 
Otherwise we may need to add a flag in GCInfo header.